### DISCUSSION TITLE
Pulling titles and links from relative file imports

### DISCUSSION THREAD

**Discussion by SundeepB** [facebook/docusaurus#9119] [Q&A] [answered]
Hello, 

I want to refer to other pages in the documentation in `.mdx` files and would like the title and the relative url pulled from the other page's frontMatter.

I was able to achieve this to a degree in https://codesandbox.io/p/sandbox/funny-https-yvw47y

The only changes from the forked default are in `docs/intro.md` and `docs/tutorial-basics/deploy-your-site.md`

In `docs/tutorial-basics/deploy-your-site.md`, I added a front matter key value

```yml
title: Deploy your site to a server
```

and in `docs/intro.md`, I added these few lines

```tsx
import {frontMatter as fm} from './tutorial-basics/deploy-your-site.md'


:::tip
<div>
There is a tutorial for "{fm.title}"
</div>
:::
```

This is able to pull the title from the linked page and when I change the title in the frontmatter in the `deploy-your-site.md` page, the value changes in `intro.md` page as well.

However, as I plan to have a lot of such links, I tried to move this functionality into a reach component at `src/components/custom/RelativeImports`

```ts
import { useEffect, useState } from "react";
import React from 'react';


export default function RelativeImport({importPath}) {
    
    
    const [title, setTitle] = useState('Unknown title')
    
    
    useEffect(() => {
        import(importPath).then(a => {
        console.log(a.frontMatter.title)
        setTitle(a.frontMatter.title)
    })
    })
    
    return (
        <div>
            My import: {title} for {importPath}
        </div>
    )
    
    
}
```

and after importing it in `intro.md`, the react component is not able to find the `deploy-your-site.md` for all the relative urls I was trying

```ts
import RelativeImport from '@site/src/components/custom/RelativeImports'

<RelativeImport importPath='@site/docs/tutorial-basics/deploy-your-site.md'/>
<RelativeImport importPath='./docs/tutorial-basics/deploy-your-site.md'/>
<RelativeImport importPath='docs/tutorial-basics/deploy-your-site.md'/>
<RelativeImport importPath='../docs/tutorial-basics/deploy-your-site.md'/>
<RelativeImport importPath='../../docs/tutorial-basics/deploy-your-site.md'/>
<RelativeImport importPath='../../../docs/tutorial-basics/deploy-your-site.md'/>
```

I get uncaught runtime errors due to modules not being found, but the site renders 

```md
My import: Unknown title for @site/docs/tutorial-basics/deploy-your-site.md
My import: Unknown title for ./docs/tutorial-basics/deploy-your-site.md
My import: Unknown title for docs/tutorial-basics/deploy-your-site.md
My import: Unknown title for ../docs/tutorial-basics/deploy-your-site.md
My import: Unknown title for ../../docs/tutorial-basics/deploy-your-site.md
My import: Unknown title for ../../../docs/tutorial-basics/deploy-your-site.md
```

I do lack the understanding of how imports would work in this case, but I wanted to check if there is a way to make this custom component work or if this has similar limitations as mentioned in the WARNING section at the bottom of https://docusaurus.io/docs/markdown-features/links ? 

Note: I checked issues and discussions if there were any other approaches/attempts, but couldn't find any. If there is a different way to achieve this, I'd be happy to give it a try. I noticed [this feature request](https://docusaurus.io/feature-requests/p/automated-link-text) that sounds similar to what I am trying here, but not sure how similar or different this feature is.

**Comment by slorber** [comment, marked as answer]
I think what you try to achieve is best achieved by creating a remark plugin.
You'll have to learn how to do this on the MDX site. 

I recommend that you implement this on MDX 2 directly, coming in Docusaurus v3, currently in alpha.
